geld.lua

```diff
@@ -0,0 +1,133 @@
+-- Gelds animals by default. Optional args: -toggle or -ungeld
+-- Written by Josh Cooper(cppcooper) on 2019-12-10, last modified: 2020-02-23
+utils ={}
+utils = require('utils')
+local validArgs = utils.invert({
+    'unit',
+    'toggle',
+    'ungeld',
+    'help',
+    'find'
+
+})
+local args = utils.processArgs({...}, validArgs)
+local help = [====[
+
+geld
+====
+Geld allows the user to geld and ungeld animals.
+
+Valid options:
+    unit <id> - Performs action on the provided unit id, this is optional.
+                If this argument is not given, the highlighted unit is
+                used instead.
+
+    toggle    - Changes the geld status to the opposite of its current state
+    ungeld    - Sets the geld status to false
+    help      - Shows this help information
+
+]====]
+
+unit=nil
+
+if args.unit then
+    id=tonumber(args.unit)
+    if id then
+        for _,unit_ in pairs (df.global.world.units.active) do
+            if unit.id == id then
+                unit=unit_
+            end
+        end
```

Comment:
I'm replacing this with just `unit = df.unit.find(id)`, which is faster (since it does a binary search). Incidentally, your implementation mixed up `unit` and `unit_` on line 37